tests.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+import datetime
+from note import Note
+from note import ReminderNote
+from note import NotesManager
+from note import TextNote
+
+'''Testing the 'display' feature of the Base class- Note 
+    I will use 'assert' to confirm the display function actually include ID, content and time created 
+'''
+
+def test_note_display():
+    content = "This is a display test to see if the function works properly"
+    note = Note(content)
+    result = note.display()
+    assert "ID: 1" in result
+    assert content in result
+    assert "created at" in result
+
+
+def test_textnote_display():
+    content = "This is a text note display test to see if the function works properly"
+    note = TextNote(content)
+    result = note.display()
+    assert "ID: 2" in result
+    assert content in result
+    assert "created at" in result
+
+def test_remindernote_display():
+    content= "This is a reminder note display test to see if the function works properly"
+    reminder_date_and_time = "28-2-2025 9:00"
+    reminder_note=ReminderNote(content, reminder_date_and_time)
+    result = reminder_note.display()
+    assert "ID: 3" in result
+    assert content in result
+    assert "created at" in result
+    assert reminder_date_and_time in result
+
+def test_notesmanager_add_text_note():
+    manager= NotesManager()
+    note_type= "text"
+    content = "This is a text note display test to see if the function works properly"
+    manager.add_note(note_type, content)
+    assert isinstance(manager.notes[0], TextNote)
+
+
+"""
+Testing adding a reminder note through NotesManager.
+Checks that the note is added and contains the correct reminder date.
+"""
+def test_notesmanager_add_reminder_note():
+    manager= NotesManager()
+    note_type= "reminder"
+    content = "This is a reminder note display test to see if the function works properly"
+    reminder_date_and_time = "28-2-2025 9:00"
+    manager.add_note(note_type, content, reminder_date_and_time)
+    assert isinstance(manager.notes[0], ReminderNote)
+    assert manager.notes[0].content == "This is a reminder note display test to see if the function works properly"
+    assert manager.notes[0].reminder_date_and_time == reminder_date_and_time
+    
+
+"""
+Testing adding a note with an invalid note type.
+Expects an error message and verifies that no note is added.
+"""
+def test_add_invalid_note():
+    manager = NotesManager()
+    result = manager.add_note("image", "Note of type image should not be added")
+    assert result == "Please choose a note type"
+    assert len(manager.notes) == 0  # To confirm the note was not stored in the note manager. That is the number of notes should be zero

```

Comment:
What if a valid note is already in the storage? this would not work in that case because `len(manager.notes)` would be more than 0.